core/sql/ustat/hs_parser.cpp

```diff
@@ -428,25 +435,22 @@ Lng32 AddTableName( const hs_table_type type
             LM->Log(LM->msg);
          }
 
-        const char* period = strchr(hs_globals->catSch->data(), '.');
-        NAString catName(hs_globals->catSch->data(),
-                         period - hs_globals->catSch->data());
+        NAString catName(hs_globals->objDef->getCatName());
+
+        *hs_globals->hstogram_table = getHistogramsTableLocation(hs_globals->catSch->data(), FALSE);
+
+        *hs_globals->hsintval_table = getHistogramsTableLocation(hs_globals->catSch->data(), FALSE);
+
         NABoolean isHbaseOrHive = HSGlobalsClass::isHbaseCat(catName) ||
                                   HSGlobalsClass::isHiveCat(catName);
 
-        *hs_globals->hstogram_table = 
-            getHistogramsTableLocation(hs_globals->catSch->data(), FALSE);
-        if (isHbaseOrHive)
+        if (isHbaseOrHive) {
           hs_globals->hstogram_table->append(".").append(HBASE_HIST_NAME);
-        else
-          hs_globals->hstogram_table->append(".HISTOGRAMS");
-
-        *hs_globals->hsintval_table = 
-            getHistogramsTableLocation(hs_globals->catSch->data(), FALSE);
-        if (isHbaseOrHive)
           hs_globals->hsintval_table->append(".").append(HBASE_HISTINT_NAME);
-        else
+        } else {
+          hs_globals->hstogram_table->append(".HISTOGRAMS");
           hs_globals->hsintval_table->append(".HISTOGRAM_INTERVALS");
+        }
```

Comment:
In trafodion, the test isHbaseOrHive will always be true. So we will always
be using SB_HISTOGRAMS and SB_HISTOGRAM_INTERVALS.  The other two names are
use by SQ.

Maybe later on, we can do a cleanup.

On Mon, Sep 21, 2015 at 1:49 PM, Roberta Marton <notifications@github.com>
wrote:

> In core/sql/ustat/hs_parser.cpp
> <https://github.com/apache/incubator-trafodion/pull/87#discussion_r40009684>
> :
>
> >            hs_globals->hsintval_table->append(".HISTOGRAM_INTERVALS");
> > +        }
>
> I currently don't have a working SQL, but are histogram table names
> HISTOGRAMS or SB_HISTOGRAMS?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/87/files#r40009684>.
>



-- 
Regards, --Qifan